File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/transactions/object/UaWriteNodeTransaction.java
package org.opcfoundation.webserver.service.transactions.object;

import org.eclipse.milo.opcua.stack.core.StatusCodes;
import org.eclipse.milo.opcua.stack.core.UaRuntimeException;
import org.opcfoundation.webserver.service.transactions.base.UaWriteTransaction;
import org.opcfoundation.webapi.service.types.WriteContext;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public class UaWriteNodeTransaction extends UaWriteTransaction {
    public UaWriteNodeTransaction(
            WriteContext context,
            List<Integer> handleIds)
    {
        super(context, handleIds);
    }

    @Override
    public CompletableFuture<Void> execute()
    {
        return CompletableFuture.supplyAsync(()-> buildErrorResponse(new UaRuntimeException(StatusCodes.Bad_NotWritable)));
    }
}

File: opcua-webservice-node/source/src/server/service/transactions/reactiveobject/UaWriteNodeTransaction.ts
import { StatusCodes } from "opcua-webapi";
import { makeUaStatusCode } from "opcua-webapi-ts";
import { WriteContext } from "../../..";
import { UaWriteTransaction } from "../base/UaWriteTransaction";

export class UaWriteNodeTransaction extends UaWriteTransaction {
    constructor(
        context: WriteContext,
        handleIds: Array<number>,
    ) {
        super(context, handleIds);
    }

    override async execute(): Promise<void> {
        this._results.length = 0;

        for (const _handleId of this.handleIds) {
            this._results.push(makeUaStatusCode(StatusCodes.BadNotWritable));
        }
    }
}